python/lsst/ts/salkafka/component_producer_set.py

```diff
@@ -315,6 +317,18 @@ def make_argument_parser():
             "Example 'my.kafka:9000'. "
             "Required unless --validate or --show-schema are specified.",
         )
+        parser.add_argument(
+            "--username",
+            default=None,
+            dest="username",
+            help="Kafka username for SASL authentication.",
+        )
+        parser.add_argument(
+            "--password",
+            default=None,
+            dest="password",
+            help="Kafka password for SASL authentication.",
```

Comment:
In all of this review I will assume that a user must either provide both of these arguments or neither. If that is false then many of my comments are irrelevant. So, assuming that....

Please expand the help to say that both or neither must be provided.